### PR TITLE
refactor(translate): zero-deserialization envelope for all cross-format paths

### PR DESCRIPTION
**Pull request by makosblade**
## Summary

- Eliminates `json.Unmarshal`/`json.Marshal` round-trips from all cross-format request emit paths and non-streaming response translators, restoring the `RequestEnvelope`'s zero-copy contract end-to-end
- Introduces a `jsonWriter` builder (GoF Builder pattern) that handles structural JSON concerns (commas, braces, escaping) so emit functions read as declarative field mappings — gjson reads on the immutable body, `jsonWriter` for output, `gjson.Result.Raw` for transplanting unchanged subtrees
- Extracts shared cross-format primitives (`parseDataURL`, `extractThoughtSignature`, `toolResultContentGJSON`) to eliminate DRY violations across emit files
- Removes `ensureSrc()`, the `src map[string]any` field, `shallowClone`, and `clampOutputTokens` — narrow `json.Unmarshal` remains only for per-tool-schema sanitization (Gemini/OpenAI allow-list filtering)
- Replaces `rewriteModelField` in openaicompat with a single `sjson.SetBytes` call and rewrites `applySystemReminderToBody` to use gjson/sjson
- Adds `make precommit` / `make install-hooks` for local CI-equivalent checks (gofmt + vet + build + test)

## Behavioral audit against main

Full four-agent audit of every cross-format path. Regressions found and fixed:

| # | Issue | Severity | Fix |
|---|-------|----------|-----|
| 1 | `tool_choice: "none"` no longer suppressed tools in Anthropic emit | HIGH | Check before writing tools; skip both tools and tool_choice |
| 2 | Missing tool `description` → malformed JSON (`"description":,`) | HIGH | Guard with existence check before writing key |
| 3 | Null `stop` value → malformed JSON (`"stop_sequences":}`) | MEDIUM | Move Key() inside type branches |
| 4 | Response translators swallowed invalid upstream JSON errors | HIGH | `gjson.ValidBytes` check at top of each function |
| 5 | Invalid tool_call arguments silently substituted `{}` in Gemini path | MEDIUM | `gjson.Valid` check with error propagation |

Benign differences (not regressions, not fixed):

| # | Issue | Assessment |
|---|-------|-----------|
| 6 | Assistant messages always emit `"content":null` when key absent from source | Benign — Anthropic requires content on all messages; only affects pathological absent-key (not null) inputs |
| 7 | Unrecognized content-part types produce `[]` instead of original array passthrough | Benign — only affects malformed input with unknown block types |
| 8 | `parseDataURL` is stricter about non-base64 data URLs | Improvement — old code incorrectly treated `data:mime,raw` as base64 |

## Test plan

- [x] All 162 pre-existing translate tests pass unchanged
- [x] 80 new cross-format matrix tests cover every request + response conversion path, stop-reason mapping, usage translation, thought-signature preservation, error contracts, and edge cases (missing description, null stop, tool_choice none, invalid JSON, invalid tool args)
- [x] New tests verified against `origin/main` — all pass on both old and new code (except edge-case tests that catch the regressions above, which were written test-first/TDD)
- [x] `go vet`, `go build ./...` clean
- [x] openaicompat tests pass (4 tests)

🤖 Generated with [Claude Code](https://claude.com/claude-code)

### CODE REVIEW

File: internal/translate/emit_gemini.go [hunk 1]
```diff
@@ -158,586 +138,921 @@ func copyMap(m map[string]any) map[string]any {
...
-	}
+		if params := fn.Get("parameters"); params.Exists() {
+			var schema any
+			if err := json.Unmarshal([]byte(params.Raw), &schema); err == nil {
+				schema = sanitizeSchemaForGemini(schema)
+				if schemaBytes, err := json.Marshal(schema); err == nil {
+					dw.Key("parameters")
+					dw.RawBytes(schemaBytes)
+				}
+			}
```
[the remark this answers is not in the document]
Comment (steventohme) [reply internal/translate/emit_gemini.go:429]: should def do this will fail if not cus of some mcps

**Review by steventohme — DISMISSED** [review]
